OCPBUGS-20194: Stop other control plane components on non-recovery hosts during etcd restore procedure

nalhadef opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Main

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s):
<!--- Specify the version or versions of OpenShift your PR applies to. -->
4.11 - 4.15

Issue:
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->
[Link to issue](https://issues.redhat.com/browse/OCPBUGS-20194)

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->
https://70568--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state#dr-scenario-2-restoring-cluster-state_dr-restoring-cluster-state

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

nalhadef commented:
/jira refresh

nalhadef commented:
@palonsoro 

Hi, Pablo.

Please check out [this preview](https://70568--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state#dr-scenario-2-restoring-cluster-state_dr-restoring-cluster-state) and let me know whether I covered all your concerns.

Thanks!

Neal

palonsoro commented:
Hello.

Among the requirements in [OCPBUGS-20194](https://issues.redhat.com/browse/OCPBUGS-20194), it only covered the one about steps 7 and 4e (which was 4d at the time of reporting, but is now 4e).

However, it does not cover the main point of the bug report, which is the required changes in step 4 about `kube-controller-manager-pod.yaml` and `kube-scheduler-pod.yaml`. 

Please review that point. It contains the requested set of changes on step 4. Note that the steps might look somewhat similar to some of the current ones, but the names are not equal (basically we now request to stop 2 components and we have to also stop another 2 with similar steps).

nalhadef commented:
Thanks, Pablo. I think (hope) I made the changes you asked for here. The PR
preview generation process is taking a very long time, but I clicked the
link
<https://70568--ocpdocs-pr.netlify.app/openshift-enterprise/latest/backup_and_restore/control_plane_backup_and_restore/disaster_recovery/scenario-2-restoring-cluster-state#dr-scenario-2-restoring-cluster-state_dr-restoring-cluster-state>
and
the changes seem to be there. i hope i got it all covered this time.

Thanks,

Neal

On Mon, Jan 22, 2024 at 6:18 AM Pablo Alonso Rodriguez <
***@***.***> wrote:

> Hello.
>
> Among the requirements in OCPBUGS-20194
> <https://issues.redhat.com/browse/OCPBUGS-20194>, it only covered the one
> about steps 7 and 4e (which was 4d at the time of reporting, but is now 4e).
>
> However, it does not cover the main point of the bug report, which is the
> required changes in step 4 about kube-controller-manager-pod.yaml and
> kube-scheduler-pod.yaml.
>
> Please review that point. It contains the requested set of changes on step
> 4. Note that the steps might look somewhat similar to some of the current
> ones, but the names are not equal (basically we now request to stop 2
> components and we have to also stop another 2 with similar steps).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/70568#issuecomment-1903791365>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ553AVGH5QVBU5HNX3YPZDIPAVCNFSM6AAAAABCCFEO7GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBTG44TCMZWGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nalhadef commented:
/retest

palonsoro commented:
@nalhadef You correctly added steps for kube-controller-manager.yaml, which is ok (thanks) but steps for kube-scheduler are still missing. Please add those as well.

Thanks in advance

nalhadef commented:
/retest

nalhadef commented:
/retest

jeana-redhat commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -90,7 +90,36 @@ $ sudo crictl ps | grep kube-apiserver | egrep -v "operator|guard"
+sudo mv /etc/kubernetes/manifests/kube-scheduler-pod.yaml /tmp
+----
+
+.. Verify that the Kubernetes Scheduler containers are stopped by using the following command:
++
+[source,terminal]
+sudo crictl ps | grep kube-scheduler | egrep -v "operator|guard"
+---- 
```
```suggestion
.. Verify that the Kubernetes Scheduler containers are stopped by using the following command:
+
[source,terminal]
----
sudo crictl ps | grep kube-scheduler | egrep -v "operator|guard"
---- 
```

palonsoro commented:
All the points in the bug description are now properly addressed.

Thank you very much @nalhadef !!!

/lgtm

nalhadef commented:
You're welcome/thanks,  Pablo!!

On Fri, Jan 26, 2024 at 3:48 AM Pablo Alonso Rodriguez <
***@***.***> wrote:

> All the points in the bug description are now properly addressed.
>
> Thank you very much @nalhadef <https://github.com/nalhadef> !!!
>
> /lgtm
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/70568#issuecomment-1911685303>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZZMJUZMKLWIW4Y2AL3YQNUWXAVCNFSM6AAAAABCCFEO7GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJRGY4DKMZQGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nalhadef commented:
Approved by the reporter and sent to the possible QE.

geliu2016 commented:
/lgtm

nalhadef commented:
/label peer-review-needed

nalhadef commented:
Thanks, Ge Liu!

On Sun, Jan 28, 2024 at 9:40 PM Ge Liu ***@***.***> wrote:

> /lgtm
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/70568#issuecomment-1913864850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ2K7PS5Q6YLDF7WB6TYQ4D2LAVCNFSM6AAAAABCCFEO7GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJTHA3DIOBVGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

dfitzmau commented:
/remove-label peer-review-needed

/label peer-review-in-progress

dfitzmau commented:
Hi @nalhadef . I'm just about to review this PR, but would you be OK to tick the QE review box and remove "4.162 from the impacted versions?

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -90,7 +90,37 @@ $ sudo crictl ps | grep kube-apiserver | egrep -v "operator|guard"
+----
+$ sudo crictl ps | grep kube-controller-manager | egrep -v "operator|guard"
+----
+The output of this command should be empty. If it is not empty, wait a few minutes and check again.
```
The IBM SG states:

![Screenshot from 2024-01-29 12-23-34](https://github.com/openshift/openshift-docs/assets/57954076/88342245-a9b8-48c1-b429-f285a203fae4)


How about: 

```suggestion
Empty output is the desired result from running the command. If the output is not empty, wait a few minutes and check again.
```
WDYT?

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -90,7 +90,37 @@ $ sudo crictl ps | grep kube-apiserver | egrep -v "operator|guard"
+----
+$ sudo crictl ps | grep kube-scheduler | egrep -v "operator|guard"
+---- 
+The output of this command should be empty. If it is not empty, wait a few minutes and check again.
```
Same feedback as previous comment.

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -90,7 +90,37 @@ $ sudo crictl ps | grep kube-apiserver | egrep -v "operator|guard"
+---- 
+The output of this command should be empty. If it is not empty, wait a few minutes and check again.
+
+.. Move the etcd data directory to a different location with the following command:
```
"Example" should be mentioned as the command is just that based on the `tmp` target location. The preceeding text does not suggest a location limitation.

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -144,13 +174,15 @@ starting kube-scheduler-pod.yaml
 ----
 +
+
+The cluster-restore.sh must show that etcd, kube-apiserver, kube-controller-manager and kube-scheduler pods are stopped and then started at the end of the restore process.
```
```suggestion
The `cluster-restore.sh` script must show that etcd, kube-apiserver, kube-controller-manager, and kube-scheduler pods are stopped and then started at the end of the restore process.
```
JUst want to check that etcd, kube-apiserver, kube-controller-manager, and kube-scheduler all refer to "pods"?

palonsoro replied:
> Just want to check that etcd, kube-apiserver, kube-controller-manager, and kube-scheduler all refer to "pods"?

All of them are pods, mentioned by the script

palonsoro replied:
And it would be best if you write "The `cluster-restore.sh` script output must show..."

ousleyp replied:
I'm not sure if you tried to make this update, but it is still showing the original formatting and text (`The cluster-restore.sh must show...`)

Please update this line to something like this:
```
The output of the `cluster-restore.sh` script must show...
```

My understanding is that verbatim pod names should be in backticks [1], so I would suggest the following formatting (this suggestion includes a couple minor grammatical tweaks as well):

```suggestion
The output of the `cluster-restore.sh` script must show that the `etcd`, `kube-apiserver`, `kube-controller-manager`, and `kube-scheduler` pods are stopped and then started at the end of the restore process.
```
[1]
> Use backticks for all non-GUI "system items", including:
> 
> - Inline commands, operations, literal values, variables, parameters, settings, flags, environment variables, user input
> - System term/item, user names, unique or example names for individual API objects/resources (e.g., a pod named mypod), daemon, service, or software package
> - RPM packages
> - File names or directory paths

(source: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting)

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -209,7 +241,7 @@ $ sudo systemctl restart kubelet.service
 ====
 
-.. Get the list of current CSRs:
+.. Get the list of current CSRs, by running the following command:
```
I think you can drop the comma as the sentence is short. With the comment, a subject is missing in the dependent clause.

ousleyp replied:
+1, no comma necessary

dfitzmau commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -228,22 +260,22 @@ csr-zhhhp   3m8s   kubernetes.io/kube-apiserver-client-kubelet   system:servicea
 <2> A pending `node-bootstrapper` CSR.
 
-.. Review the details of a CSR to verify that it is valid:
+.. Review the details of a CSR to verify that it is valid by running the following command:
```
I do not see "CSR" expanded in the document. Maybe expand it's first occurence in the doc?

dfitzmau commented:
Hi @nalhadef . I added some minor comments inline. Please ping me if you have any queries.

/remove-label peer-review-in-progress

/label peer-review-done

nalhadef commented:
/label merge-review-needed

If approved, please merge. Thanks.

nalhadef commented:
/retest

nalhadef commented:
/retest

ousleyp commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -620,14 +652,14 @@ AllNodesAtLatestRevision
 ----
 +
-Verify all nodes are updated to the latest revision.
+Verify all nodes are updated to the latest revision by running the following command.
```
This should end in `:` instead of `.`

ousleyp commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -645,14 +677,14 @@ AllNodesAtLatestRevision
 ----
 +
-Verify all nodes are updated to the latest revision.
+Verify all nodes are updated to the latest revision by using the following command.
```
This should end in `:` instead of `.`, and I would suggest s/by using/by running for consistency

ousleyp commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -265,7 +297,7 @@ $ sudo crictl ps | grep etcd | egrep -v "operator|etcd-guard"
 ----
 
-.. From the recovery host, verify that the etcd pod is running.
+.. From the recovery host, verify that the etcd pod is running by running the following command:
```
Backticks for `etcd`?

palonsoro replied:
In this case, as it refers to the etcd pod, it would make sense. But not as a general guideline on how to write "etcd" in the docs.

ousleyp commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -252,7 +284,7 @@ $ oc adm certificate approve <csr_name>
 . Verify that the single member control plane has started successfully.
 
-.. From the recovery host, verify that the etcd container is running.
+.. From the recovery host, verify that the etcd container is running by running the following command:
```
Backticks for `etcd`?

ousleyp reviewed (commented):
Hi @nalhadef , it looks like some changes that you thought you made after peer review didn't get pushed. Please review my comments, some of which reflect Darragh's original comments - I apologize for any redundancy. Thank you so much!

palonsoro commented on modules/dr-restoring-cluster-state.adoc:
```diff
@@ -90,7 +90,37 @@ $ sudo crictl ps | grep kube-apiserver | egrep -v "operator|guard"
 The output of this command should be empty. If it is not empty, wait a few minutes and check again.
 
-.. Move the etcd data directory to a different location:
+.. Move the existing Kubernetes Controller Manager pod file out of the kubelet manifest directory by using the following command:
```
I agree with using `kube-controller-manager`, as it matches the name of the pod we are talking about and would make it easier to find for the user who is dealing with this.

However, we would also to use `kube-scheduler` and `kube-apiserver` with the same notation, for full consistency.

nalhadef commented:
@ousleyp and @pablintino, I've addressed all your comment. The content seems to have push correctly today and all the changes should be visible in the preview (at least they are right now). Thanks again for looking them over.

ousleyp commented:
Hey @nalhadef , thanks for making the changes. :) If you're ready for merge review again, please put the merge-review-needed label back on the PR so that it goes into the queue. Much appreciated!

nalhadef commented:
/label merge-review-needed

nalhadef commented:
Done. Thanks.

On Tue, Jan 30, 2024 at 3:59 PM Pan Ousley ***@***.***> wrote:

> Hey @nalhadef <https://github.com/nalhadef> , thanks for making the
> changes. :) If you're ready for merge review again, please put the
> merge-review-needed label back on the PR so that it goes into the queue.
> Much appreciated!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/70568#issuecomment-1917884111>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ27YZ6ZIF5RGDNCL53YRFNMRAVCNFSM6AAAAABCCFEO7GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJXHA4DIMJRGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

palonsoro commented:
@nalhadef Regarding this line:
```
An installed command-line interface for CRI-compatible container runtimes (`crictl`).
```

This shouldn't be listed as a requirement. Control plane nodes can only use RHCOS, which already ships the right version of this tool and it is unsupported (and difficult) to remove or replace it.

nalhadef commented:
removed, thanks, Pablo!

On Wed, Jan 31, 2024 at 4:34 AM Pablo Alonso Rodriguez <
***@***.***> wrote:

> @nalhadef <https://github.com/nalhadef> Regarding this line:
>
> An installed command-line interface for CRI-compatible container runtimes (`crictl`).
>
> This shouldn't be listed as a requirement. Control plane nodes can only
> use RHCOS, which already ships the right version of this tool and it is
> unsupported (and difficult) to remove or replace it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/70568#issuecomment-1918724937>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AT7EEZ56VQ727IDAZ6VKD73YRIF2LAVCNFSM6AAAAABCCFEO7GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJYG4ZDIOJTG4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mburke5678 commented:
/cherrypick enterprise-4.11

mburke5678 commented:
/cherrypick enterprise-4.12

mburke5678 commented:
/cherrypick enterprise-4.13

mburke5678 commented:
/cherrypick enterprise-4.14

mburke5678 commented:
/cherrypick enterprise-4.15
